web-locks/non-fully-active.tentative.https.html

```diff
@@ -0,0 +1,70 @@
+<!DOCTYPE html>
+<meta charset=utf-8>
+<title>Web Locks API: Non-fully-active documents</title>
+<link rel=help href="https://wicg.github.io/web-locks/">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="resources/helpers.js"></script>
+
+<div></div>
+
+<script>
+  function createNonFullyActiveIframe(src) {
+    const iframe = document.createElement("iframe");
+    document.body.appendChild(iframe);
+    const { navigator, DOMException, postMessage } = iframe.contentWindow;
+    iframe.remove();
+    return { iframe, navigator, DOMException, postMessage };
+  }
+
+  promise_test(async t => {
+    const { navigator, DOMException } = createNonFullyActiveIframe();
+    const p = navigator.locks.request("foo", () => { });
+    await promise_rejects_dom(t, "InvalidStateError", DOMException, p, "Request should explicitly fail");
+  }, "request() on non-fully-active document must fail");
+
+  promise_test(async t => {
+    const { navigator, DOMException } = createNonFullyActiveIframe();
+    const p = navigator.locks.query();
+    await promise_rejects_dom(t, "InvalidStateError", DOMException, p, "Query should explicitly fail");
+  }, "query() on a non-fully-active document must fail");
+
+  promise_test(async t => {
+    const { navigator, DOMException, postMessage } = createNonFullyActiveIframe();
+
+    const p = navigator.locks.request("-", () => { });
+    await promise_rejects_dom(t, "InvalidStateError", DOMException, p, "Request should explicitly fail");
+  }, "request()'s fully-active check happens earlier than name validation");
+
+  promise_test(async t => {
+    const { iframe, navigator, DOMException } = createNonFullyActiveIframe();
+    document.body.append(iframe);
+    t.add_cleanup(() => iframe.remove());
+
+    assert_not_equals(navigator, iframe.contentWindow.navigator, "Navigator object changes");
```

Comment:
This behavior is presumably specified outside of web-locks itself, right? Can you add a comment with a link?

Reply:
```suggestion
    // Appending should create a new browsing context with a new Navigator object
    // https://html.spec.whatwg.org/multipage/iframe-embed-object.html#the-iframe-element:insert-an-element-into-a-document
    // https://html.spec.whatwg.org/multipage/system-state.html#the-navigator-object:associated-navigator
    assert_not_equals(navigator, iframe.contentWindow.navigator, "Navigator object changes");
```

Does this change look good?

Reply:
Yep, LGTM. (This was mostly stemming from my ignorance about how detached iframes behave)